### PR TITLE
Permeation test case modification

### PR DESCRIPTION
**Pull request by jhdark**
## Proposed changes

Made some small changes to the test case to ensure the model works. i.e re-calling the `create_solver()` function to add the boundary conditions, defined after initialising, to the formulation and using the ds created in the HydrogenTransportProblem class. Furthermore, the test is compared against an analytical solution.

## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [x] New tests

## Checklist

- [x] Black formatted
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: test/test_permeation_problem.py [hunk 1]
```diff
@@ -76,13 +70,12 @@ def siverts_law(T, S_0, E_S, pressure):
     )
     bc_outgas = dirichletbc(Constant(my_mesh.mesh, PETSc.ScalarType(0)), right_dofs, V)
     my_model.boundary_conditions = [bc_sieverts, bc_outgas]
-
-    final_time = 50
-
-    # log.set_log_level(log.LogLevel.INFO)
+    my_model.create_solver()
```
Comment (RemDelaporteMathurin) [inline test/test_permeation_problem.py:73]: if this is what is needed for the test to work, then I'd like the call to `.create_solver()` to be removed from `.initialise()` for now.

File: test/test_permeation_problem.py [hunk 2]
```diff
@@ -96,23 +89,43 @@ def siverts_law(T, S_0, E_S, pressure):
...
-
-        mobile_xdmf.write_function(u, t)
-
         # update previous solution
         mobile_H.prev_solution.x.array[:] = u.x.array[:]
 
     mobile_xdmf.close()
 
+    # analytical solution
+    S = 4.02e21 * exp(-1.04 / F.k_B / float(temperature))
```
Comment (RemDelaporteMathurin) [inline test/test_permeation_problem.py:106]: The pre-exp and activation energy are duplicated (defined here and on line 67
```python
surface_conc = siverts_law(T=temperature, S_0=4.02e21, E_S=1.04, pressure=100)
```

Same goes for the temperature and mesh length. 

Can you store all these in variables to avoid code duplication?

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
Looks like a needed addition to this test.

Could you add docstrings to the test?

File: test/test_permeation_problem.py [hunk 3]
```diff
@@ -96,23 +91,39 @@ def siverts_law(T, S_0, E_S, pressure):
...
+    summation = np.sum(
+        (-1) ** n_array
+        * np.exp(-((np.pi * n_array) ** 2) * float(my_model.D) / L**2 * times),
+        axis=0,
+    )
+    analytical_flux = P_up**0.5 * permeability / L * (2 * summation + 1)
+
+    flux_values = np.array(np.abs(flux_values))
+    normalised_analytical_flux = analytical_flux / analytical_flux.max()
+    normalised_compute_flux = flux_values / flux_values.max()
```
Comment (RemDelaporteMathurin) [inline test/test_permeation_problem.py:120]: No we don't want to normalise anything! We may miss some buggy behaviour. Imagine the normalised curves match but not the absolute ones

File: test/test_permeation_problem.py [hunk 4]
```diff
@@ -96,23 +91,39 @@ def siverts_law(T, S_0, E_S, pressure):
...
+
+    diff = np.abs(normalised_analytical_flux - normalised_compute_flux)
+    error = diff.mean()
+
+    assert error < 0.001
```
Comment (RemDelaporteMathurin) [inline test/test_permeation_problem.py:125]: This is the absolute mean difference. We may want to go for the relative difference instead.

Imagine a case where the computed flux is zero at all times, and the analytical flux is always below 0.001 H/m2/s

Then the absolute difference will be below 0.001 (because 0.001 - 0 = 0.001) and yet the code doesn't work!

**Review by RemDelaporteMathurin — CHANGES_REQUESTED** [review]
Thanks for addressing the comments, just a few things to change in the test

**Comment by RemDelaporteMathurin** [comment]
Confirmed docs builds are failing due to a change outside of festim.

I'll try fixing it in another PR, ignore for now
